### PR TITLE
Add PyMEEP recipe

### PR DESCRIPTION
**Pull request by ChristopherHogan**

### CODE REVIEW

**Comment by ChristopherHogan** [comment]
@conda-forge-admin, please rerender

**Comment by ChristopherHogan** [comment]
Travis is timing out at 1h 15m for the OSX build. Running py 2.7, 3.5 and 3.6 builds with all tests for each takes a long time. Is there anything I can do besides skipping some tests? Why aren't the builds of separate python versions run in a build matrix instead of one giant job? Thanks.
@conda-forge/core

File: recipes/pymeep/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - toolchain
+    - gcc
```
Comment (scopatz) [inline recipes/pymeep/meta.yaml]: Please use the new compilers to to try to improve your build times: https://conda-forge.org/docs/meta.html?#compilers

Comment (CJ-Wright) [reply recipes/pymeep/meta.yaml]: Please use the new syntax. It seems that `libctlgeom` is missing from this build, which is causing the errors.

Comment (ChristopherHogan) [reply recipes/pymeep/meta.yaml]: It finds `libctlgeom` with the old compilers. With the new syntax, the `PATH` doesn't seem to be set up correctly. `configure` can't find any dependencies (hdf5, gsl, zlib, harminv, etc.).
```
checking for harminv_get_freq_error in -lharminv... no
configure: WARNING: harminv support is disabled
checking mpb.h usability... no
checking mpb.h presence... no
checking for mpb.h... no
checking for cos in -lm... yes
checking for cblas_dgemm in -lgslcblas... no
checking for gsl_sf_bessel_Jn in -lgsl... no
configure: WARNING: Missing GNU GSL library...Bessel-function field initialization will not be supported.
checking for deflate in -lz... no
configure: WARNING: zlib is required for HDF5!
checking for H5Pcreate in -lhdf5... no
configure: WARNING: Couldn't find the HDF5 library!!  Switching to --without-hdf5.
```

Comment (ChristopherHogan) [reply recipes/pymeep/meta.yaml]: Also, the instructions @scopatz linked to above say each dependency should be built with the new compilers. This package has a lot of dependencies that don't seem to use the new compilers (e.g., [hdf5](https://github.com/conda-forge/hdf5-feedstock/blob/master/recipe/meta.yaml).

Comment (CJ-Wright) [reply recipes/pymeep/meta.yaml]: Conda-Forge has not moved over to the new compilers, yet. The new syntax is just a wrapper around the current `toolchain` setup. However, when we do make the migration over to the new compilers we need the new syntax.

I think to fix the problem with that is causing the trace above is from not having a `host` section.
see: https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#host

Comment (ChristopherHogan) [reply recipes/pymeep/meta.yaml]: Using the new compilers did not improve the build times. It still times out on MacOS. Do I have to skip tests to ensure that the jobs complete in under 1h 20m, or can you recommend an alternative?

**Comment by scopatz** [comment]
Yeah, I would try skipping integration tests to start with

**Comment by ChristopherHogan** [comment]
This is green and ready for review. Thanks.
@conda-forge/core

**Pull request merged**